### PR TITLE
#CU-31md7qg  added staking rewards to runtime with some basic test setup

### PR DESCRIPTION
**Pull request by dzmitry-lahoda**
Signed-off-by: Dzmitry Lahoda <dzmitry@lahoda.pro>

## Description
- added staking rewards to dali
- added test runtime 
## Checklist

- [x] I have updated the cargo docs to reflect changes made by this PR _(if applicable)_
- [x] I have updated the `book/` to reflect changes made by this PR _(if applicable)_

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-31md7qg Add pallet into Dali  runtime](https://app.clickup.com/t/31md7qg)

**Review by vivekvpandya — APPROVED** [review]

**Review by andor0 — APPROVED** [review]

File: runtime/dali/src/lib.rs [hunk 1]
```diff
@@ -862,6 +862,24 @@ impl crowdloan_rewards::Config for Runtime {
 	type Time = Timestamp;
 }
 
+parameter_types! {
+	pub const StakingRewardsPalletId : PalletId = PalletId(*b"stk_rwrd");
+}
+
+impl pallet_staking_rewards::Config for Runtime {
+	type Event = Event;
+	type Share = u128;
```
Comment (vimukthi-git) [inline runtime/dali/src/lib.rs:871]:
```suggestion
	type Share = Balance;
```
May be?

File: frame/composable-tests-helpers/src/test/currency.rs [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+/// # use pallet_lending::currency::Currency;
+///
+/// type ACOIN = Currency<12345, 10>;
+/// type BCOIN = Currency<54321, 12>;
+///
+/// let one_hundred_a_tokens = ACOIN::units(100);
+/// let one_value_of_b = BCOIN::ONE;
+/// ```
+#[derive(Debug, Clone, Copy, TypeInfo)]
+pub struct Currency<const ID: u128, const EXPONENT: u8> {}
```
Comment (vimukthi-git) [inline frame/composable-tests-helpers/src/test/currency.rs:17]: :+1:

**Review by vimukthi-git — DISMISSED** [review]
LGTM

**Comment by KaiserKarel** [comment]
Can you guys rebase?

File: frame/staking-rewards/README.md [hunk 3]
```diff
@@ -52,23 +52,21 @@ A pool is where users put tokens and get shares of rewards or other powers.
 
 Any protocol can have one or more pools. The main characteristic of a pool is staked asset identifier(token).
 
-Each pool is governed through a protocol that created it.
```
Comment (hussein-aitlahcen) [inline frame/staking-rewards/README.md:55]: Why remove the 'that'?

File: frame/staking-rewards/README.md [hunk 4]
```diff
@@ -52,23 +52,21 @@ A pool is where users put tokens and get shares of rewards or other powers.
...
 
-Can/must user time lock tokens to get, possibly increased, `share`. What is the penalty for early unlock if there is a lock
-  
-If share decays with time until lock ends and extending time starts from zero elapsed.
+Can/must user time lock tokens to get, possibly increased, `share`. What is the penalty for early unlock if there is a lock.
 
-Or if the share does not decay with time and extending time subsumes old-time already passed in a lock (user gets smaller remaining time with higher share).
+By default share does not decay with time and extending time subsumes old-time already passed in a lock (user gets smaller remaining time with higher share).
 
 Owner account of the pool. Usually other protocol or its "owner".
```
Comment (hussein-aitlahcen) [inline frame/staking-rewards/README.md:68]: Keep 'it'.

**Review by andor0 — APPROVED** [review]

**Review by hussein-aitlahcen — APPROVED** [review]

**Review by vivekvpandya — APPROVED** [review]

**Pull request merged**